### PR TITLE
Fix decoding NSData from an NSCoder

### PR DESCRIPTION
**Pull request by gmilos**
The new test case shows how `NSData` decoding could fail. Fix included.

### CODE REVIEW

**Comment by millenomi** [comment]
@swift-ci please test

File: TestFoundation/TestNSData.swift [hunk 1]
```diff
@@ -4140,5 +4141,19 @@ extension TestNSData {
         XCTAssertEqual(data[data.startIndex.advanced(by: 1)], 0xFF)
 #endif
     }
+
+    func test_nskeyedarchiving() {
+        let bytes: [UInt8] = [0xd, 0xe, 0xa, 0xd, 0xb, 0xe, 0xe, 0xf]
+        let data = NSData(bytes: bytes, length: bytes.count)
+
+        let archiver = NSKeyedArchiver()
+        let key = "data"
```
Comment (spevans) [inline TestFoundation/TestNSData.swift]: `key` is never used.

File: TestFoundation/TestNSData.swift [hunk 2]
```diff
@@ -4140,5 +4141,19 @@ extension TestNSData {
...
+        data.encode(with: archiver)
+        let encodedData = archiver.encodedData
+
+        let unarchiver = NSKeyedUnarchiver(forReadingWithData: encodedData)
```
Comment (spevans) [inline TestFoundation/TestNSData.swift]: According to https://developer.apple.com/documentation/foundation/nskeyedunarchiver/1410862-init the method is `init(forReadingWith data: Data)` so needs a small adjustment.

**Comment by gmilos** [comment]
@spevans thanks for the comments. Fixed both. Botched rebase, apologies.

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
Is this relevant still? `as? Data` should work for both NSData and Data values now.

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by kevints** [comment]
@millenomi @gmilos is out this week but i think this patch is still relevant - if not the implementation the test case should still be merged

**Comment by spevans** [comment]
I just ran the test against `master` and it worked OK so its probably worth adding the just the test to catch any later regressions.

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Comment by millenomi** [comment]
@swift-ci please test and merge

**Pull request merged**